rest/route/annotations.go

```diff
@@ -319,3 +320,107 @@ func (h *annotationByTaskPutHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 	return responder
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PUT /rest/v2/tasks/{task_id}/created_ticket
+
+// this api will be used by teams who set up their own web hook to file a ticket, to send
+// us the information for the ticket that was created so that we can store and display it.
+
+type createdTicketByTaskPutHandler struct {
+	taskId    string
+	execution *int
+	user      gimlet.User
+	ticket    *model.APIIssueLink
+	sc        data.Connector
+}
+
+func makeCreatedTicketByTask(sc data.Connector) gimlet.RouteHandler {
+	return &createdTicketByTaskPutHandler{
+		sc: sc,
+	}
+}
+
+func (h *createdTicketByTaskPutHandler) Factory() gimlet.RouteHandler {
+	return &createdTicketByTaskPutHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *createdTicketByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+	h.taskId = gimlet.GetVars(r)["task_id"]
+	if h.taskId == "" {
+		return gimlet.ErrorResponse{
+			Message:    "task ID cannot be empty",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	// for now, tickets will be created for a specific task execution only
+	// and will not be visible on other executions
+	if r.URL.Query().Get("execution") == "" {
```

Comment:
For the Put annotation route we allow users to not specify execution and then just use the latest execution, don't we? Why don't we allow that here?

Reply:
I was a bit conflicted about it. I ultimately decided to make the execution required because when the user clicks on the file ticket button, it will be for a task on a specific execution. A different execution for that task might not even have a failed status (and therefore no tab for these to show up in). The user would expect the ticket to show up where they clicked the button. I don't see a good reason where using the latest execution would ever be helpful, so I'd assume that if someone leaves the execution blank, it is a mistake and not for a specific reason.